### PR TITLE
Arpcache Impl w/ bugs

### PR DESCRIPTION
**Pull request by davepetrov**
Not exactly sure why im getting errors when i set variables to appropriate fields before sending packets? - ask alina and prey

### CODE REVIEW

File: cs144_lab3/router/sr_arpcache.c [hunk 1]
```diff
@@ -12,70 +12,145 @@
...
-    if (request->times_sent >=5){
-        printf("[DEBUG] Resent ARP Req. 5 times -> Send ICMP-Host-Unreachable (Type3, Code 1)");
+    if (request->times_sent >= ARP_REQUEST_MAXIMUM){
+        printf("[DEBUG] Resent ARP Req. %d times -> Send ICMP-Host-Unreachable (Type3, Code 1)\n", ARP_REQUEST_MAXIMUM);
         /* Destination Host Unreachable (Type 3, code 1): Tried sending 5 
         or more arp reqs to the next-hop IP without response*/
         struct sr_packet *packet = (struct sr_packet *)request->packets;
         while (packet != NULL){
-            /* TODO: Send ICMP message w/ corresponding Type=3 and code=1 */
+            /* TODO: Ask Alina if she implemented? Send ICMP message w/ corresponding Type=3 and code=1 */
```
Comment (alinabuzila) [inline cs144_lab3/router/sr_arpcache.c:66]: yep this is implemented ready to test

**Comment by alinabuzila** [comment]
Try using memset on those lines that error

File: cs144_lab3/router/sr_arpcache.c (hunk 1 above)
Comment (davepetrov) [reply cs144_lab3/router/sr_arpcache.c:66]: merge your code. ill make a pr after u merge that code in so i can test after

File: cs144_lab3/router/sr_arpcache.c [hunk 2]
```diff
@@ -73,7 +73,84 @@ void handle_arp_request_queue(struct sr_instance *sr, struct sr_arpreq* request,
...
+    
+    /* Build corresponding Ethernet header*/
+    create_ethernet_header((struct sr_ethernet_hdr *) arpBroadcastRequest, interface);
+    create_arp_header((struct sr_arp_hdr*)(arpBroadcastRequest+sizeof(struct sr_ethernet_hdr)), interface, request);
+
+    /* Send ARP request */
+    printf("[DEBUG] Sending ARP request\n");
+    sr_send_packet(sr, arpBroadcastRequest, sizeof(struct sr_ethernet_hdr)+sizeof(struct sr_arp_hdr), interface->name);
+    request->times_sent+= 1;
+
```
Comment (kingpreyansh) [inline cs144_lab3/router/sr_arpcache.c:93]: why aren't you updating the req. sent time to the current time?

Comment (davepetrov) [reply cs144_lab3/router/sr_arpcache.c:93]: will fix thx

Comment (kingpreyansh) [inline cs144_lab3/router/sr_arpcache.c:93]: Also, you should probably free the memory after this call since you malloc it for arpBroadcastRequest

Comment (davepetrov) [reply cs144_lab3/router/sr_arpcache.c:102]: fixed

**Pull request merged**